embrace-gradle-plugin/src/main/java/io/embrace/android/gradle/plugin/instrumentation/visitor/ApplicationClassAdapter.kt

```diff
@@ -0,0 +1,37 @@
+package io.embrace.android.gradle.plugin.instrumentation.visitor
+
+import com.android.build.api.instrumentation.ClassContext
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.MethodVisitor
+
+/**
+ * Visits the Application class and returns an [ApplicationMethodAdapter] for the onCreate method.
+ */
+class ApplicationClassAdapter(
```

Comment:
This current implementation only works if an `Application` class exists and if the `onCreate` function has an override. I think that would be a reasonable approach but we will need to document this on the docs site (along with the new gradle DSL property)